ksqldb-functional-tests/src/test/resources/query-validation-tests/row-offset-and-partition.json

```
@@ -119,6 +150,622 @@
+      }
+    },
+    {
+      "name": "INSERT INTO ROWOFFSET should fail",
+      "properties": {
+        "ksql.rowpartition.rowoffset.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM input (id INT KEY, val INT) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "INSERT INTO input (id, val, rowoffset) VALUES (1, 2, 3);"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "Failed to insert values into 'INPUT'. Inserting into column `ROWOFFSET` is not allowed."
+      }
+    },
+    {
+      "name": "Select ROWPARTITION and ROWOFFSET columns with aliases",
+      "properties": {
+        "ksql.rowpartition.rowoffset.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM input (id INT KEY, foo STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM output AS SELECT id, ROWPARTITION AS rp, ROWOFFSET AS ro, ROWTIME AS rt FROM input;"
+      ],
+      "inputs": [
+        {"topic":  "test_topic", "key": 1, "value": {"foo": "a"}, "timestamp": 5},
+        {"topic":  "test_topic", "key": 2, "value": {"foo": "b"}, "timestamp": 6},
+        {"topic":  "test_topic", "key": 3, "value": {"foo": "c"}, "timestamp":  7}
+      ],
+      "outputs": [
+        {"topic":  "OUTPUT", "key": 1, "value": {"RP": 0, "RO": 0, "RT": 5}, "timestamp": 5},
+        {"topic":  "OUTPUT", "key": 2, "value": {"RP": 0, "RO": 1, "RT": 6}, "timestamp": 6},
+        {"topic":  "OUTPUT", "key": 3, "value": {"RP": 0, "RO": 2, "RT": 7}, "timestamp": 7}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "ID INT KEY, RP INT, RO BIGINT, RT BIGINT"}
+        ]
+      }
+    },
+    {
+      "name": "Select ROWPARTITION/ROWOFFSET from CSAS",
+      "properties": {
+        "ksql.rowpartition.rowoffset.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM a (id INT KEY, foo STRING) WITH (kafka_topic='a', value_format='JSON');",
+        "CREATE STREAM b AS SELECT * from a;",
+        "CREATE STREAM c AS SELECT *, ROWPARTITION as rp, ROWOFFSET as ro from b;"
+      ],
+      "inputs": [
+        {"topic":  "a", "key": 1, "value": {"foo": "foo"}, "timestamp": 5},
+        {"topic":  "a", "key": 2, "value": {"foo": "bar"}, "timestamp": 5},
+        {"topic":  "a", "key": 3, "value": {"foo": "baz"}, "timestamp": 5}
+      ],
+      "outputs": [
+        {"topic":  "C", "key": 1, "value": {"foo": "foo", "RP": 0, "RO": 0}, "timestamp": 5},
+        {"topic":  "C", "key": 2, "value": {"foo": "bar", "RP": 0, "RO": 1}, "timestamp": 5},
+        {"topic":  "C", "key": 3, "value": {"foo": "baz", "RP": 0, "RO": 2}, "timestamp": 5}
+      ],
+      "post": {
+        "sources": [
+          {"name": "C", "type": "stream", "schema": "id INT KEY, foo STRING, RP INT, RO BIGINT"}
+        ]
+      }
+    },
+    {
+      "name": "Select ROWPARTITION/ROWOFFSET from CT",
```

```suggestion
      "name": "Select ROWPARTITION/ROWOFFSET with aliases - table",
```
and rename `Select ROWPARTITION and ROWOFFSET columns with aliases` above to `Select ROWPARTITION/ROWOFFSET with aliases - stream` for consistency.